examples/step-44/step-44.cc

```diff
@@ -31,10 +31,13 @@
 #include <deal.II/base/tensor.h>
 #include <deal.II/base/timer.h>
 #include <deal.II/base/work_stream.h>
-
 #include <deal.II/dofs/dof_renumbering.h>
 #include <deal.II/dofs/dof_tools.h>
 
+// This header gives us the functionality to store
+// data at quadrature points
+#include <deal.II/base/quadrature_point_data.h>
```

Comment:
there are several classes there. `CellDataStorage` is related to storage only, whereas there are a couple related to transfer: `TransferableQuadraturePointData` and `ContinuousQuadratureDataTransfer`. I was considering splitting them https://github.com/dealii/dealii/pull/2552#issuecomment-217630237 but then we agreed to keep them in one place. We can still split them if you prefer.